src/OpenTelemetry.AutoInstrumentation.Native/integration_loader.h

```diff
@@ -19,29 +19,37 @@ using json = nlohmann::json;
 // in the OTEL_DOTNET_AUTO_INTEGRATIONS_FILE environment variable
 void LoadIntegrationsFromEnvironment(
     std::vector<IntegrationMethod>& integrationMethods,
-    const std::vector<WSTRING>& enabledIntegrationNames,
-    const std::vector<WSTRING>& disabledIntegrationNames);
+    const std::vector<WSTRING>& enabledTraceIntegrationNames,
+    const std::vector<WSTRING>& disabledTraceIntegrationNames,
+    const std::vector<WSTRING>& enabledLogIntegrationNames,
+    const std::vector<WSTRING>& disabledLogIntegrationNames);
 
 // LoadIntegrationsFromFile loads the integrations from a file
 void LoadIntegrationsFromFile(
     const WSTRING& file_path,
     std::vector<IntegrationMethod>& integrationMethods,
-    const std::vector<WSTRING>& enabledIntegrationNames,
-    const std::vector<WSTRING>& disabledIntegrationNames);
+    const std::vector<WSTRING>& enabledTraceIntegrationNames,
+    const std::vector<WSTRING>& disabledTraceIntegrationNames,
+    const std::vector<WSTRING>& enabledLogIntegrationNames,
+    const std::vector<WSTRING>& disabledLogIntegrationNames);
 
 // LoadIntegrationsFromFile loads the integrations from a stream
 void LoadIntegrationsFromStream(
     std::istream& stream,
     std::vector<IntegrationMethod>& integrationMethods,
-    const std::vector<WSTRING>& enabledIntegrationNames,
-    const std::vector<WSTRING>& disabledIntegrationNames);
+    const std::vector<WSTRING>& enabledTraceIntegrationNames,
+    const std::vector<WSTRING>& disabledTraceIntegrationNames,
+    const std::vector<WSTRING>& enabledLogIntegrationNames,
+    const std::vector<WSTRING>& disabledLogIntegrationNames);
 
 namespace
 {
     void IntegrationFromJson(const json::value_type& src,
                          std::vector<IntegrationMethod>& integrationMethods,
-                         const std::vector<WSTRING>& enabledIntegrationNames,
-                         const std::vector<WSTRING>& disabledIntegrationNames);
+                         const std::vector<WSTRING>& enabledTraceIntegrationNames,
+                         const std::vector<WSTRING>& disabledTraceIntegrationNames,
+                         const std::vector<WSTRING>& enabledLogIntegrationNames,
+                         const std::vector<WSTRING>& disabledLogIntegrationNames);
```

Comment:
This is getting very nasty. I think it is time to refactor and introduce a `Integrations` class so that we can change the signatures to something like:

```
Integrations LoadIntegrationsFrom*(inputs...)
```

or even instead of functions create static methods in `Integrations` class so that the are named e.g. `FromJson` or `LoadJson`


We can do it in a separate PR to make it easier for reviewing.

Reply:
I agree that it could be done better, tracked under #1482